src/prefect/workers/base.py

```diff
@@ -871,17 +871,19 @@ async def _submit_run_and_capture_errors(
             deployment = await self._client.read_deployment(flow_run.deployment_id)
         if deployment and deployment.concurrency_limit:
             limit_name = f"deployment:{deployment.id}"
-            concurrency_limit = deployment.concurrency_limit
             concurrency_ctx = concurrency
+
+            # ensure that the global concurrency limit is available
+            # and up-to-date before attempting to acquire a slot
+            await self._client.upsert_global_concurrency_limit_by_name(
+                limit_name, deployment.concurrency_limit
+            )
```

Comment:
same note as above - i think this logic belongs in `create_deployment` / `update_deployment`

Reply:
We want it there, too, but doing it here prevents a few issues related to keeping the deployment and limit in sync. For example, a user can edit a deployment's concurrency limit from the UI. Logic here prevents us from needing to change server-side APIs to handle that until we do the larger change to use a GCL foreign key on deployments.